In [1]:
import pandas as pd
import pprint

In [32]:
datadir = '/home/xavier/data/'
acmg_file = 'ACMG_Conditions_OMIM.csv'
clinvar_file = 'variant_summary.txt'

In [33]:
acmg = pd.read_csv(datadir + acmg_file)

In [4]:
clinvar = pd.read_csv(datadir + clinvar_file, sep='\t', lineterminator='\n')

/home/xavier/miniconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (18) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [5]:
clinvar.shape

(809509, 31)

In [34]:
acmg.shape

(59, 2)

# This is a matching exercise

We are trying to match the disease name in ACMG Recos to the 

+ The files we are working from:
    + [ACMG Recommendations for Reporting of Incidental Findings in Clinical Exome and Genome Sequencing](https://www.ncbi.nlm.nih.gov/clinvar/docs/acmg/)
    + [Clinvar Variants Summary](ftp://ftp.ncbi.nlm.nih.gov/pub/clinvar/tab_delimited/)

## So I anticipate my stumbling block to be that I don't efficiently match disease names. Some workarounds

+ Instead of matching name, match ID ... e.g. MIM ID
+ Since there are only 59 'Disease Names' in ACMG, instead of matching by column, match by each disease name as a variable. 
    + > So you can judge how many variants for a keyword
    
### In all events, faster to just work with the Phenotype ID, List

In [7]:
phenos =  clinvar.loc[:,'PhenotypeList']

In [8]:
phenos

0                Spastic paraplegia 48, autosomal recessive
1                Spastic paraplegia 48, autosomal recessive
2                Spastic paraplegia 48, autosomal recessive
3                Spastic paraplegia 48, autosomal recessive
4                                   Galloway-Mowat syndrome
5                                   Galloway-Mowat syndrome
6           Mitochondrial complex I deficiency;not provided
7           Mitochondrial complex I deficiency;not provided
8                        Mitochondrial complex I deficiency
9                        Mitochondrial complex I deficiency
10        Inborn genetic diseases;Mitochondrial complex ...
11        Inborn genetic diseases;Mitochondrial complex ...
12                       Mitochondrial complex I deficiency
13                       Mitochondrial complex I deficiency
14                       Mitochondrial complex I deficiency
15                                HFE INTRONIC POLYMORPHISM
16                                HFE IN

In [9]:
a = 'Adenomatous polyposis coli'
b = 'polyposis coli'
c = 'coli'

In [10]:
onebyone = clinvar[clinvar['PhenotypeList'].str.contains(c, na=False)]
onebyone.shape

(701, 31)

In [11]:
onebytwo = clinvar[clinvar['PhenotypeList'].str.contains(b, na=False)]
onebytwo.shape

(4, 31)

In [12]:
onebythree = clinvar[clinvar['PhenotypeList'].str.contains(a, na=False)]
onebythree.shape

(4, 31)

In [31]:
byid = clinvar[clinvar['PhenotypeIDS'].str.contains('175100', na=False)]
byid.shape

(3490, 31)

## There is a lot of mismatching going on here!

+ 701 rows for 'Coli'
+ 4 rows for 'Adenomatous polyposis coli'
+ 3490 rows for 'ID = 175100'

## ...⇧
> **HYPOTHESIS:** Looks like the smart thing would be to match by ID.

- - -
Let's write an **"ISIN"** definition that matches all the, or something like that.
These solutions both look promising:
+ https://datascience.stackexchange.com/questions/19227/create-a-new-column-based-on-two-columns-from-two-different-dataframes
+ https://stackoverflow.com/questions/27474921/compare-two-columns-using-pandas?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa
